### ISSUE TITLE
Microsoft specific handling of variadic macros not supported

### ISSUE THREAD

**Issue by Bertk** [SonarOpenCommunity/sonar-cxx#843]
The sample [variadic_macros.cpp](https://msdn.microsoft.com/en-us/library/ms177415.aspx) from MSDN fails with _com.sonar.sslr.api.RecognitionException_
```
#define MACRO(s, ...) printf(s, __VA_ARGS__)

int main() {
    MACRO("hello, world\n");
}
```
 The syntax tree is created for the next code sample but error messages are shown:
```
#define EMPTY
#define MACRO(s, ...) printf(s, __VA_ARGS__)

int main() {
    MACRO("error\n", EMPTY);  // would cause error C2059, except VC++ 
                              // suppresses the trailing comma
}
```
e.g.
```
08:22:26.021 [AWT-EventQueue-0] ERROR CxxPreprocessor - Mismatch: expected ',' got: ')'
```

**Comment by Bertk** [comment]
@guwirth I have a solution proposal for this (b7c588adf441dce5807a59e4c5fc32789997e677). Please review and send me your feedback.
There is still an issue with the BOOST_PP_STRINGIZE which I want to analyze. But this doesn't belog to variadic macros ...
```
INFO: [D:\Workspaces\xxx\boost\include\boost\mpl\list.hpp:36]: cannot parse included filename: BOOST_PP_STRINGIZE(boost/mpl/list/list20.hpp)'
INFO: [D:\Workspaces\xxx\boost\include\boost\mpl\list.hpp:36]: cannot find the sources for '#   include BOOST_PP_STRINGIZE(boost/mpl/list/AUX778076_LIST_HEADER)'
INFO: [D:\Workspaces\xxx\boost\include\boost\mpl\vector.hpp:36]: cannot parse included filename: BOOST_PP_STRINGIZE(boost/mpl/vector/vector20.hpp)'
INFO: [D:\Workspaces\xxx\boost\include\boost\mpl\vector.hpp:36]: cannot find the sources for '#   include BOOST_PP_STRINGIZE(boost/mpl/vector/AUX778076_VECTOR_HEADER)'

```

**Comment by Bertk** [comment]
This issue is solved with PR #853
